### PR TITLE
fix(views/core): iteration over None type

### PR DESCRIPTION
**Pull request by rasta-rocket**
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by villebro — COMMENTED** [review]
I agree with this change (can happen if `adhoc_filters` is defined but has a value that isn't a list). @rasta-rocket can you update the PR and mark it as non-draft so we can get it merged?

**Comment by rasta-rocket** [comment]
@villebro done
(thx for relaunching this by the way 😉 )

File: superset/views/core.py [hunk 1]
```diff
@@ -959,7 +959,7 @@ def save_or_overwrite_slice(
             slc = Slice(owners=[g.user] if g.user else [])
 
         form_data["adhoc_filters"] = self.remove_extra_filters(
-            form_data.get("adhoc_filters", [])
+            (form_data.get("adhoc_filters", []) or [])
```
Comment (villebro) [inline superset/views/core.py:962]: nit (I thought I added this comment, but it seems to have disappeared into the ether); I think this should work just as well to handle where `adhoc_filters` is set to `None`:
```suggestion
            form_data.get("adhoc_filters") or []
```

Comment (rasta-rocket) [reply superset/views/core.py:962]: I don't really remember why I did that :sweat_smile:
But agree :+1:

**Comment by villebro** [comment]
Thanks @rasta-rocket ! Merging when the last tests pass

**Review by villebro — APPROVED** [review]

**Pull request merged**